### PR TITLE
Support for FHIR Patch

### PR DESCRIPTION
**Pull request by ZuSe**
Related to https://github.com/beda-software/fhir-py/issues/68

So we don't forget it. 

Further discussion in this thread pls

### CODE REVIEW

File: fhirpy/base/lib.py [hunk 1]
```diff
@@ -119,6 +119,9 @@ def execute(self, path, method='post', **kwargs):
         return self._do_request(method, path, **kwargs)
 
     def _do_request(self, method, path, data=None, params=None):
+        if method == 'patch':
+            self.extra_headers['Content-Type'] = 'application/json-patch+json'
```
Comment (ruscoder) [inline fhirpy/base/lib.py:123]: not sure that it's good idea to override client's headers on this step.

**Comment by ruscoder** [comment]
Hi @ZuSe. Thanks for the contribution. For now it's quite not ready PR.

File: fhirpy/base/lib.py (hunk 1 above)
Comment (BSVogler) [reply fhirpy/base/lib.py:123]: The request is constructed only just before it is sent and is needed. So where else could it be?

Comment (ruscoder) [reply fhirpy/base/lib.py:123]: Mutating self.extra_headers affect all future requests. It's better to override this header further
```
r = requests.request(method, url, json=data, headers=headers)
```

**Pull request closed without merging**